src/lib/plugins/GraphQLPlugin/server/HttpClient.ts

```diff
@@ -0,0 +1,159 @@
+import 'isomorphic-fetch'
+import { forEach, toPairs } from 'lodash'
+import * as qs from 'querystring'
+import { InjectionContext, InjectionClient } from '../../../core/InjectionClient'
+
+export class HttpError implements Error {
+  readonly status: number
+
+  constructor(status: number) {
+    this.status = status
+  }
+
+  get message() {
+    return `Http ${this.status}`
+  }
+
+  get name() {
+    return 'HTTPError'
+  }
+}
+
+export type HttpMethod = 'GET' | 'PUT' | 'PATCH' | 'POST' | 'DELETE'
+
+export interface FetchUrl {
+  baseUrl: string
+  path: string
+  query?: {}
+}
+
+export interface RequestOpts {
+  url: string | FetchUrl
+  body?: any
+  headers?: Record<string, string>
+  responseType?: ResponseType
+  bodyType?: BodyType
+}
+
+export interface ResponseType {
+  headers: Record<string, string>
+  convert: (x: Response) => Promise<any> | undefined
+}
+
+export interface BodyType {
+  headers: Record<string, string>
+  convert: (x: any) => any
+}
+
+export const BodyType = {
+  FormData: {
+    convert: (body: Record<string, string>) => {
+      const form = new FormData()
+      forEach(body, (value, key) => {
+        form.append(key, value)
+      })
+      return form
+    },
+    headers: { 'Content-Type': 'application/x-www-form-urlencoded' }
+  },
+  Json: {
+    convert: JSON.stringify,
+    headers: { 'Content-Type': 'application/json' }
+  },
+  Empty: {
+    convert: () => undefined,
+    headers: {}
+  }
+}
+
+export const ResponseType = {
+  Json: {
+    convert: (x: Response) => x.json(),
```

Comment:
You may want to handle the case where a 201 is returned

Reply:
Should we treat 201s as a special case, or put the responsibility on the developer to specify a different response type?

Reply:
Hmmmm unsure. You could argue either way I think, I'm OK with leaving it as is.